src/Nuvemshop/Provider.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace SocialiteProviders\Nuvemshop;
+
+use GuzzleHttp\Exception\GuzzleException;
+use GuzzleHttp\RequestOptions;
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
+
+/**
+ * @see https://tiendanube.github.io/api-documentation/authentication
+ */
+class Provider extends AbstractProvider
+{
+    public const IDENTIFIER = 'NUVEMSHOP';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $scopes = [
+        'read_content',
+        'write_content',
+        'read_products',
+        'write_products',
+        'read_customers',
+        'write_customers',
+        'read_orders',
+        'write_orders',
+        'read_coupons',
+        'write_coupons',
+        'write_scripts',
+        'write_shipping'
+    ];
+
+    /**
+     * The separating character for the requested scopes.
+     *
+     * @var string
+     */
+    protected $scopeSeparator = ' ';
+
+    /**
+     * Returns the configured user id that we're authenticating with
+     *
+     * @return string
+     */
+    private function getClientId()
```

Comment:
This is useless: the `clientId` is already available as class property. Please remove it and update the code accordingly.